src/test/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/net/RejectingExecutor.java

```diff
@@ -0,0 +1,84 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.net;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * An executor which throws an {@link RejectedExecutionException} any time it is asked to execute a task
+ * This class is useful for testing certain branches of our net code
+ */
+class RejectingExecutor implements ExecutorService {
+
+    @Override
+    public void shutdown() {}
+
+    @Override
+    public List<Runnable> shutdownNow() {
+        return null;
+    }
+
+    @Override
+    public boolean isShutdown() {
+        return false;
+    }
+
+    @Override
+    public boolean isTerminated() {
+        return false;
+    }
+
+    @Override
+    public boolean awaitTermination(long l, TimeUnit timeUnit) throws InterruptedException {
+        return false;
+    }
+
+    @Override
+    public <T> Future<T> submit(Callable<T> callable) {
+        return null;
+    }
+
+    @Override
+    public <T> Future<T> submit(Runnable runnable, T t) {
+        return null;
+    }
+
+    @Override
+    public Future<?> submit(Runnable runnable) {
+        return null;
+    }
+
+    @Override
+    public <T> List<Future<T>> invokeAll(Collection<? extends Callable<T>> collection) throws InterruptedException {
+        return null;
+    }
+
+    @Override
+    public <T> List<Future<T>> invokeAll(Collection<? extends Callable<T>> collection, long l, TimeUnit timeUnit)
+            throws InterruptedException {
+        return null;
+    }
+
+    @Override
+    public <T> T invokeAny(Collection<? extends Callable<T>> collection)
+            throws InterruptedException, ExecutionException {
+        return null;
+    }
+
+    @Override
+    public <T> T invokeAny(Collection<? extends Callable<T>> collection, long l, TimeUnit timeUnit)
+            throws InterruptedException, ExecutionException, TimeoutException {
+        return null;
+    }
+
+    @Override
+    public void execute(Runnable runnable) {
+        throw new RejectedExecutionException("REJECTED");
+    }
+}
```

Comment:
We are missing a newline at the end of the file ?

Reply:
Fixed